### DISCUSSION TITLE
Using the GFX library, I am not seeing SPI signals using ST7701s device with ESP32?

### DISCUSSION THREAD

**Discussion by puddletowntom** [moononournation/Arduino_GFX#312] [General]
I was trying to run the HelloWorld example using your GFX library with the ST7701s IC driver. I am using the official Espressif dev board (ESP32-S3-WROOM-1 N8R8). From your wiki article on device declaration, I used the code below.

I was able to get HSYNC, VSYNC, PCLK, DE signals when looking at the osciliscope but i am not seeing an SPI signals. Eg pin 48, 47 have no signals appearing on the scope? Interestingly, when i try many of the other examples from your wiki I see the SPI signals. I tested with ST7789 and I can see the SPI signals. 

https://github.com/moononournation/Arduino_GFX/wiki/Dev-Device-Declaration

I noticed from the example from Makerfabs in the link below that there are compatibility issues with version is 1.3.1 and to use the latest version. However even when i use the latest version and test the makerfab device declaration I still dont see any SPI lines on the oscilloscope. I am surprised by this given many of the other examples seem to work? Has the device declaration from ST7701s become outdated?

https://github.com/Makerfabs/ESP32-S3-Parallel-TFT-with-Touch-4inch

 ```#include <Arduino_GFX_Library.h>
#define GFX_BL 38
Arduino_DataBus *bus = new Arduino_SWSPI(
    GFX_NOT_DEFINED /* DC */, 39 /* CS */,
    48 /* SCK */, 47 /* MOSI */, GFX_NOT_DEFINED /* MISO */);

// option 1:
// Uncomment for 4" rect display
Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
    18 /* DE */, 17 /* VSYNC */, 16 /* HSYNC */, 21 /* PCLK */,
    4 /* R0 */, 3 /* R1 */, 2 /* R2 */, 1 /* R3 */, 0 /* R4 */,
    10 /* G0 */, 9 /* G1 */, 8 /* G2 */, 7 /* G3 */, 6 /* G4 */, 5 /* G5 */,
    15 /* B0 */, 14 /* B1 */, 13 /* B2 */, 12 /* B3 */, 11 /* B4 */,
    1 /* hsync_polarity */, 10 /* hsync_front_porch */, 8 /* hsync_pulse_width */, 50 /* hsync_back_porch */,
    1 /* vsync_polarity */, 10 /* vsync_front_porch */, 8 /* vsync_pulse_width */, 20 /* vsync_back_porch */);
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
    480 /* width */, 480 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush */,
    bus, GFX_NOT_DEFINED /* RST */, st7701_type1_init_operations, sizeof(st7701_type1_init_operations));```

**Comment by moononournation** [comment]
There are only a very short period have SPI signal while calling begin() function.

**Reply by puddletowntom** [reply]
Yes, the SPI signals are only at the beginning. I actually ended up figuring out what I was doing wrong. I was using my own custom hardware and the reset pin for the LCD was held low. The coding examples I was using, werent setting the reset pin perhaps because the hardware on some of these devices have it high. It took a bit of figuring out but I just compared my signals directly to a working board. Thank you for your response. That reset pin is always something people should not forget about.
